test/PushNotificationTests/LRPTests.cpp

```diff
@@ -0,0 +1,178 @@
+            VERIFY_IS_TRUE(TP::IsPackageRegistered_DynamicDependencyDataStore());
+            VERIFY_IS_TRUE(TP::IsPackageRegistered_DynamicDependencyLifetimeManager());
+            VERIFY_IS_TRUE(TP::IsPackageRegistered_PushNotificationsLongRunningTask());
+            return true;
+        }
+
+        TEST_METHOD_CLEANUP(MethodUninit)
+        {
+            VERIFY_IS_TRUE(TP::IsPackageRegistered_WindowsAppSDKFramework());
+            VERIFY_IS_TRUE(TP::IsPackageRegistered_DynamicDependencyDataStore());
+            VERIFY_IS_TRUE(TP::IsPackageRegistered_DynamicDependencyLifetimeManager());
+            VERIFY_IS_TRUE(TP::IsPackageRegistered_PushNotificationsLongRunningTask());
+
+            return true;
+        }
+
+        TEST_METHOD(LaunchLRP_FromCoCreateInstance)
+        {
+            VERIFY_SUCCEEDED(::CoInitializeEx(nullptr, COINITBASE_MULTITHREADED));
+
+            auto scopeExit = wil::scope_exit(
+                [&]() { VERIFY_NO_THROW(CoUninitialize()); });
+
+            VERIFY_NO_THROW(winrt::create_instance<INotificationsLongRunningPlatform>(_uuidof(NotificationsLongRunningPlatform), CLSCTX_ALL));
+
+            VerifyLRP_IsRunning(true);
+
+            // LRP should be up for 30 seconds. We don't expect any app to be tracked at this point.
+            Sleep(30000);
+
+            // Verify the LRP is not running.
+            VerifyLRP_IsRunning(false);
+        }
+
+        TEST_METHOD(LaunchLRP_FromStartupTask)
+        {
+            STARTUPINFO startupInfo = { 0 };
+            wil::unique_process_information processInfo;
+
+            ZeroMemory(&startupInfo, sizeof(startupInfo));
+            startupInfo.cb = sizeof(startupInfo);
+
+            // Build the Solution OutDir path where the Startup Task exe is.
+            auto startupTaskExePath = Test::FileSystem::GetSolutionOutDirPath();
+            startupTaskExePath /= Test::Packages::PushNotificationsLongRunningTask::c_PackageDirName;
+            startupTaskExePath += L".Msix";
+            startupTaskExePath /= L"msix";
+            startupTaskExePath /= L"PushNotificationsLongRunningTask.StartupTask.exe";
+
+            BOOL wasProcessCreated = CreateProcess(
+                startupTaskExePath.c_str(),
+                nullptr, // command line options
+                nullptr, // process attributes
+                nullptr, // thread attributes
+                FALSE,   // inherit handles
+                NORMAL_PRIORITY_CLASS, // creation flags
+                nullptr, // lpEnvironment
+                nullptr, // current directory for the process
+                &startupInfo,
+                &processInfo
+            );
+
+            if (wasProcessCreated == FALSE)
+            {
+                VERIFY_SUCCEEDED(GetLastError());
+            }
+
+            // Wait for the process to come up and be captured in the snapshot from verification step.
+            Sleep(1000);
```

Comment:
Please avoid this. Can you eliminate the sleep and make the test more predictable?

Reply:
One way to make it predictable is to poll the process launch till a timeout(like 5s).

Reply:
I tried without the Sleep and the test fails because the LRP hasn't launched yet at the time of the first snapshot. It seems like there is a small overhead between launching startup task and the launching LRP.